scipy/linalg/_solvers.py

```diff
@@ -80,6 +80,20 @@ def solve_sylvester(a, b, q):
     True
 
     """
+    for ind, _ in enumerate((a, b)):
+        if not np.equal(*_.shape):
+            raise ValueError("Matrix {} should be square.".format("ab"[ind]))
+    # Shape consistency check
+    if a.shape[0] != q.shape[0] or b.shape[0] != q.shape[1]:
+        raise ValueError("Matrix q should have shape (a.shape[0], b.shape[0])")
+
+    # Accomodate empty a
+    if a.size == 0 or b.size == 0:
+        a_n = np.eye(2, dtype=a.dtype)
+        b_n = np.eye(2, dtype=b.dtype)
+        q_n = np.eye(2, dtype=q.dtype)
+        dt = solve_sylvester(a_n, b_n, q_n).dtype
```

Comment:
Solving dummy arguments is too heavy for quick exits. Instead you can use `numpy.result_type` and check if it is still integer then cast result to `np.float64`. Same with the case below.

Reply:
This PR follows other examples (mea culpa). FWIW, I'm not sure result_type matches the linalg behavior. (uint8-s may combine into f32 afair, while linalg may differ). 

Also FWIW my reasoning was this. The whole point of this size-0 business is to simplify user looping or otherwise generating samples, where a single edge case produces a size-0 array and an hour-long simulation crashes.  when everything else is O(N**3), this single edge case costs nothing in comparison. So easier to just parrot what non-trivial work does and move on.

Reply:
should I check only `np.int64` or all the types int `np.int32`, ... Or is there a way to include all these types ?

Reply:
> FWIW, I'm not sure result_type matches the linalg behavior. (uint8-s may combine into f32 afair, while linalg may differ).

It should be OK but in case you have some doubts, then you can use the `scipy.linalg.lapack.get_lapack_funcs` or `scipy.linalg.lapack.get_blas_funcs` to get the exact typecode for the resulting type. That is guaranteed to match.